Sources/SwiftProtobuf/_MessageLayout.swift

```diff
@@ -155,7 +155,19 @@ import Foundation
 
     /// The function type for the generated function that is called to retrieve the "message" layout
     /// of a map entry.
-    public typealias MapEntryLayout = (_ token: TrampolineToken) -> StaticString
+    public typealias MapEntryLayout = (_ token: TrampolineToken) -> _MessageLayout
+
+    /// The function type for the generated function that is called to perform an arbitrary
+    /// operation on the elements of a map.
+    public typealias MapEntryPerformer = (
+        _ token: TrampolineToken,
+        _ field: FieldLayout,
+        _ storage: _MessageStorage,
+        _ workingSpace: _MessageStorage,
+        _ operation: TrampolineFieldOperation,
+        _ deterministicOrdering: Bool,
```

Comment:
aside - I wonder if we should pass options directly to be more future proof on things that are public? I guess that runs into problems with json/binary/textformat having different options.  maybe we need a CommonOptions for things that this sorta common code could need and to map things between them?